crates/rnote-engine/src/store/stroke_comp.rs

```diff
@@ -640,6 +640,64 @@ impl StrokeStore {
             .collect::<Vec<StrokeKey>>()
     }
 
+    pub(crate) fn keys_between(
```

Comment:
Can this function be split up into two separate ones that are only responsible for x or y? And let the keys be supplied by the user for composing them together to get the same result.

This would get rid of the complexity (match statement) in the function body and would make it more re-usable.

Reply:
I've not done exactly that for now but I got rid of the match statement.
It's replaced with multiple filters using the bools for the state of the verticalspace tool to skip unneeded selection using the lazily evaluated or operator.

I've not tried to chain the filters, that'd need for the `stroke_components` to be public (or at least the iterator over them through a getter method) + two external functions defined on the `impl Iterator<Item  = (Strokekey,...)>` (can't be a method on self).

But at that point, I can also get the iterator from the new getter, and do the filters inside `toolspage.rs` (and maybe lazily evaluate some of the values I'm for now passing as parameters to `keys_between`)

Reply:
Check out `filter_keys_intersecting_bounds()`, that's what I had in mind.

Wouldn't something like this be possible?

```rust
let keys = store.keys_unordered();
if limit_movement_vertical_border {
    let filtered = store.keys_between_vertical_position(&keys);
    // ...
} else {
    let filtered = store.keys_below_y()
    // ...
}

Reply:
But I won't block merging this if it isn't done this way, so your call

Reply:
I'll try to do the method used in `filter_keys_intersecting_bounds()` and see how that compares. 

One question I have for this (and other related operations), is that I'm not sure we're using the Rtree everywhere for queries (there's a `keys_in_bounds` method that seems like a good candidate for doing this instead, unless there's issue if we have semi infinite bounds ?)

Reply:
yeah that's true `keys_in_bounds` could be used instead!

Reply:
Ok, this is fine for me now